libs/ledger/src/executor.cpp

```diff
@@ -170,8 +170,18 @@ Executor::Result Executor::Execute(Digest const &digest, BlockIndex block, Slice
       storage_cache_->Clear();
     }
 
+    FeeManager::TransactionDetails tx_details{
```

Comment:
Is it worth making the constructor here to take the transaction object itself? This might make is easier in the future to extend?